matrix_client/api.py

```diff
@@ -774,6 +776,70 @@ def get_download_url(self, mxcurl):
         else:
             raise ValueError("MXC URL did not begin with 'mxc://'")
 
+    def media_download(self, mxcurl):
+        """ Download raw media from provided mxc URL
+
+        Args:
+            mxcurl (str): mxc media URL
+        """
+        if mxcurl.startswith('mxc://'):
+            return self._send(
+                "GET", mxcurl[6:],
+                api_path="/_matrix/media/r0/download/"
+            )
+        else:
+            raise ValueError(
+                "MXC URL '%s' did not begin with 'mxc://'" % mxcurl
+            )
+
+    def get_thumbnail(self, mxcurl, width, height, method='scale'):
+        """ Download raw media thumbnail from provided mxc URL
+
+        Args:
+            mxcurl (str): mxc media URL
+            width (int): required thumbnail width
+            height (int): required thumbnail height
+            method (str): thumb creation method. Must be
+                 in ['scale', 'crop']. Default 'scale'.
+        """
+        if method.lower() not in ['scale', 'crop']:
```

Comment:
Not sure if tolerating improper case is useful, or consistent with the rest of the API methods.